zookeeper-server/src/main/java/org/apache/zookeeper/server/Request.java

```diff
@@ -169,24 +168,16 @@ public boolean isThrottlable() {
                 && this.type != OpCode.createSession;
     }
 
-    private transient byte[] serializeData;
-
-    @SuppressFBWarnings(value = "EI_EXPOSE_REP")
     public byte[] getSerializeData() {
         if (this.hdr == null) {
             return null;
         }
-
-        if (this.serializeData == null) {
```

Comment:
so basically you are saying that caching this value is useless ?
do you know why we were caching it ?

Reply:
See https://github.com/apache/zookeeper/pull/2030, the aim is to reduce the serialization. From the prd, the memory looks more sensitive.